pkg/config/config.go

```diff
@@ -123,19 +138,47 @@ func (r *KCLRun) Transform(rl *fn.ResourceList) error {
 		if err != nil {
 			return err
 		}
-		transformedObjects = append(transformedObjects, obj)
+
+		// Check if the transformed object matches the resource rules
+		if r.MatchResourceRules(obj) {
+			transformedObjects = append(transformedObjects, obj)
+		}
 	}
 	rl.Items = transformedObjects
 	return nil
 }
 
+// MatchResourceRules checks if the given Kubernetes object matches the resource rules specified in KCLRun.
+func (r *KCLRun) MatchResourceRules(obj *fn.KubeObject) bool {
+	// check if MatchConstraints is set
+	if isEmptyStruct(r.Spec.MatchConstraints) {
+		return true
+	}
+	// if MatchConstraints.ResourceRules is not set (nil or empty), return true by default
+	if r.Spec.MatchConstraints.ResourceRules == nil || len(r.Spec.MatchConstraints.ResourceRules) == 0 {
```

Comment:
I mean that r, r.Spec, r.MatchConstraints can be nil here and you should write code  to judge it. For example,
```
if r == nil || r.Spec || r.Spec.MatchConstraints == nil {return true}
```